mat64/doc.go

```diff
@@ -112,7 +112,34 @@
 // For convenience, a matrix may be used as both a receiver and as an input, i.e.
 //  a.Pow(a, 6)
 //  v.SolveVec(a.T(), v)
-// though in many cases this will cause an allocation. However, mat64 cannot
-// detect arbitrary overlap, and so care should be taken when using matrix views.
+// though in many cases this will cause an allocation (see Element Aliasing).
 // An exception to this rule is Copy, which does not allow a.Copy(a.T()).
+//
+// Element Aliasing
+//
+// Most methods in mat64 modify receiver data. It is forbidden for the modified
+// data region of the receiver to overlap the used data area of the input
+// arguments. The exception to this rule is when the method receiver is equal to one
+// of the input arguments, as in the a.Pow(a, 6) call above, or its implicit transpose.
+//
+// If you need to read this section to understand the behavior of your program,
+// you are being clever. Don't be clever. If you must be clever, blas64 and
+// lapack64 may be used to call the behavior directly.
+//
+// When data is shared between the receiver and input, mat64 will attempt to detect
```

Comment:
My recommendation for this documentation is:

"mat64 will use the following rules to detect overlap between the receiver and one of the inputs:
 * the input implements one of the Raw methods and
 * the Raw type matches that of the receiver and
 * the address ranges of the backing data slices overlap and
 * either the strides differ or there is an overlap in the used data elements 
If such  an overlap is detected, the method will panic.

The following cases will not panic:
 * the data slices do not overlap 
 * there is pointer identity between the receiver and input values after
    the value has been untransposed if necessary

mat64 does not detect element overlap if the input does not implement a Raw method or if the Raw method differs from that of the receiver. Correct method behavior cannot be guaranteed if there is overlap.

Currently only RawMatrixer aliasing detection is supported."